practice-app/routes/exchange_rate_api.js

```diff
@@ -115,6 +132,35 @@ function calculateAverage(res, start_date, end_date, from, to) {
     });
 }
 
+function calculatePercentage(res, change_date, prev_day, from, to) {
+    const reqOptions = {'url': process.env.EXCHANGE_RATE_API_HISTORY_URL, 'qs': {'start_at': prev_day, 'end_at': change_date, 'base': from, 'symbols': to}};
+    request(reqOptions, (err, response, body) => {
+        if (err) {
+            // If something went wrong during the request, send it to the user
+            res.status(400).send({'error': '' + err});
+        } else if (response.statusCode !== 200) {
+            // If the endpoint returns an error code (like invalid from/to symbols)
+            // send it directly to the user
+            res.status(400).send(body);
+        } else {
+            const result = JSON.parse(body);
+            var perc = 0;
+            var firstValue = 0;
+            firstValue = result['rates'][moveDate(change_date,0)][to];
+            var secondValue = 0;
+            secondValue = result['rates'][moveDate(prev_day,0)][to];
+            perc = (firstValue-secondValue)/secondValue *100;
+            /*for (var i = 0; i < dateDiff(start_date, end_date); i++) {
```

Comment:
I think you should remove this part :smile:

Reply:
removed that part